src/appsignal/cli/diagnose.py

```diff
@@ -0,0 +1,390 @@
+
+        try:
+            with open(log_file_path) as log_file:
+                lines = log_file.readlines()
+                log_file_contents = lines[-10:]
+        except FileNotFoundError:
+            log_file_contents = [""]
+
+        print("Paths")
+        print("  Current working directory")
+        print(f"    Path: '{os.getcwd()}'")
+        print()
+        print("  Log directory")
+        print(f"    Path: '{log_path}'")
+        print()
+        print("  AppSignal log")
+        print(f"    Path: '{log_file_path}'")
+
+        print("    Contents (last 10 lines):")
+        for line in log_file_contents:
+            print(line.strip())
+
+    def _report_information(self) -> None:
+        print("Diagnostics report")
+
+    def _report_prompt(self) -> bool | None:
+        print("  Do you want to send this diagnostics report to AppSignal?")
+        print("  If you share this report you will be given a link to")
+        print("  AppSignal.com to validate the report.")
+        print("  You can also contact us at support@appsignal.com")
+        print("  with your support token.")
+        print()
+
+        send_report = input("  Send diagnostics report to AppSignal? (Y/n):   ")
+
+        if send_report in ["y", "yes", ""]:
+            print("Transmitting diagnostics report")
+            return True
+        if send_report in ["n", "no"]:
+            print("Not sending diagnostics information to AppSignal.")
+            return False
+        self._report_prompt()
+        return None
+
+    def _send_diagnose_report(self) -> None:
+        params = urllib.parse.urlencode(
+            {
+                "api_key": self.config.option("push_api_key"),
+                "name": self.config.option("name"),
+                "environment": self.config.option("environment"),
+                "hostname": self.config.option("hostname") or "",
+            }
+        )
+
+        endpoint = self.config.option("diagnose_endpoint")
+        url = f"{endpoint}?{params}"
+
+        response = requests.post(url, json={"diagnose": self.report})
+
+        status = response.status_code
+        if status == 200:
+            token = response.json()["token"]
+            print()
+            print(f"  Your support token: {token}")
+            print(f"  View this report:   https://appsignal.com/diagnose/{token}")
+        else:
+            print(
+                "  Error: Something went wrong while submitting the report to AppSignal."
+            )
+            print(f"  Response code: {status}")
+            print(f"  Response body: {response.text}")
+
+    def _paths_data(self) -> dict:
+        working_dir = os.getcwd() or ""
+        log_file_path = self.config.log_file_path() or ""
+        log_path = self.config.option("log_path") or os.path.dirname(log_file_path)
+
+        try:
+            with open(log_file_path) as log_file:
+                lines = log_file.readlines()
+                log_file_contents = lines[-10:]
```

Comment:
This only stores the last 10 lines of the file in the diagnose report JSON. We would like to [read up to 2MB of data](https://github.com/appsignal/appsignal-ruby/blob/5ddde58bb492984626d2dbddb292cecdfc225576/lib/appsignal/cli/diagnose/paths.rb#L74-L77).